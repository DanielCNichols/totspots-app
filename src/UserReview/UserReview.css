.userReviewItem {
  margin: 20px 0;
  width: 100%;
  box-shadow: 0 0 5px rgba(0, 0, 0, 0.7);
  padding: 5px 10px;
  border: 2px solid #064789;
}

.review p {
  margin: 5px 0;
}

.rating_info {
  margin-bottom: 15px;
}

.user_rating {
  width: 100%;
  display: flex;
  justify-content: space-between;
  margin: 10px 0;
  align-items: center;
}

.rating > span {
  color: gold;
  font-size: 1.3em;
}

.price span {
  color: green;
}

.volume {
  color: black;
}
.user_content {
  margin: 20px 0;
}

.user_content p {
  margin: 10px 0;
}

.userReview_controls {
  display: flex;
  flex-direction: column;
  width: 80%;
  margin: 0 auto;
}

.userReview_controls button {
  display: block;
  min-width: 180px;
  height: 40px;
  margin: 10px auto;
  border-radius: 17px;
  cursor: pointer;
  background-color: transparent;
  font-size: 1.1em;
}

.delete {
  border-color: #ed254e;
}

.edit {
  border-color: #064789;
}

@media screen and (min-width: 450px) {
  .userReview_controls {
    display: flex;
    justify-content: center;
  }

  .userReview_controls button {
    width: 30%;
  }

  .userReview_controls a {
    display: block;
    min-width: 30%;
    margin: 0 auto;
  }
}

@media screen and (min-width: 800px) {
  .userReview_controls {
    flex-direction: row;
  }

  .userReview_controls button {
    display: inline-block;
    min-width: 140px;
    word-wrap: break-word;
    font-size: 1em;
  }

  .userReview_controls .edit:hover {
    color: white;
    background-color: #064789;
  }

  .delete:hover {
    color: white;
    background-color: #ed254e;
  }
}
