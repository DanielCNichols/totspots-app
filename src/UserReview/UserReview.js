import React from 'react';
import VenueContext from '../VenuesContext'
import moment from 'moment';
import ApiService from '../services/api-service';
import {Link} from 'react-router-dom'

export default class UserReview extends React.Component {
  static contextType = VenueContext;

  deleteReviewRequest(reviewId, callback) {
    console.log('deleting review')
    ApiService.deleteReview(reviewId)
    .then(() => {
      callback(reviewId);
    })
    .catch((this.context.setError))
  }

  render() {
    let {userReviews} = this.props
    return (
        <li key={userReviews.id}>
          <h4>{userReviews.venue_name}</h4>
          <p>Visited {moment(userReviews.date_created).format('MMMM Do YYY')}</p>
          <div>
          <p>Your Rating: {Math.round(userReviews.starrating)}/5</p>
          <p>Your Price: {Math.round(userReviews.price)}/5</p>
          <p>Your Volume Level: {Math.round(userReviews.volume)}/5</p>
          <p>Here's what you had to say</p>
            <p>{userReviews.content}</p>
          </div>
          <Link to={`/${userReviews.id}`}><button>Edit Review</button></Link>
          <button onClick={()=> this.deleteReviewRequest(userReviews.id, this.context.deleteReview)}>DeleteReview</button>
        </li>
    );
  }
}


