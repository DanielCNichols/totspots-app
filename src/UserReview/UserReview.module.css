.userReview {
  width: 100%;
  border: 2px solid #e7e7e7;
  border-radius: 5px;
  padding: 5px;
}

.venueHeader {
  padding-top: 5px;
}

.venueHeader,
.ratingsContainer,
.about {
  padding-left: 3px;
  padding-right: 3px;
}

.ratingsContainer {
  margin-bottom: 5px;
}

.ratingElement {
  display: flex;
  justify-content: space-between;
  width: 70%;
  margin-bottom: 5px;
}

.venueHeader {
  margin-bottom: 3px;
}

.venueHeader h3 {
  margin-bottom: 3px;
  word-wrap: break-word;
}

.venueHeader p {
  font-style: italic;
}

.content {
  position: relative;
  max-width: 100%;
  border: 2px solid #e7e7e7;
  padding: 10px;
  /* Somethings */
  border-radius: 0.5rem;
  margin: 20px 0 10px 0;
  background-color: rgb(212, 231, 248);
}

.content::before {
  content: '';
  position: absolute;
  width: 0;
  height: 0;
  bottom: 100%;
  left: 0;
  border-top: 0.75rem solid transparent;
  border-right: 0.75rem solid transparent;
  border-left: 0.75rem solid transparent;
  border-top: none;
  border-bottom: 0.8rem solid rgb(212, 231, 248);
}

.userReviewControls {
  width: 100%;
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  column-gap: 10%;
  margin: 20px auto 10px auto;
}

.userReviewControls button {
  width: 100%;
  height: 3em;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  outline: none;
  background: transparent;
  border: none;
}

.userReviewControls button svg {
  font-size: 2em;
}

.userReviewControls button span {
  font-size: 0.8em;
}

#delete {
  color: rgb(175, 21, 21);
}

#edit {
  color: #064789;
}
