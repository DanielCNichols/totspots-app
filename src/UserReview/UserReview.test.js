
import React from 'react';
import ReactDOM from 'react-dom'
import UserReview from './UserReview.js'
import {MemoryRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'

const userReviews =  {
  "id": 27,
  "content": "Fusce posuere felis sed lacus. Morbi sem mauris, laoreet ut, rhoncus aliquet, pulvinar sed, nisl. Nunc rhoncus dui vel sem. Sed sagittis.",
  "price": 3,
  "starrating": 3,
  "volume": 1,
  "date_created": "2019-10-18T19:59:22.263Z",
  "venue_id": 1,
  "user_id": 13,
  "first_name": "Joanne",
  "last_name": "Hart"
}

it('renders UserReview without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <MemoryRouter>
    <UserReview userReviews={userReviews} />
  </MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div)
})

it('renders UserReview as expected', () => {
  const tree = renderer
  .create(
  <MemoryRouter>
    <UserReview name="UserReview" userReviews={userReviews}/>
  </MemoryRouter>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
})
