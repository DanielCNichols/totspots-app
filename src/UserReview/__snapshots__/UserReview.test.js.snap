// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UserReview as expected 1`] = `
<li
  className="userReviewItem"
>
  <h4 />
  <p>
    Visited 
    October 18th, 2019
  </p>
  <div
    className="rating_info"
  >
    <div
      className="user_rating"
    >
      Your Rating: 
      <span
        className="rating"
      >
        <div>
          <span>
            â˜…
          </span>
          <span>
            â˜…
          </span>
          <span>
            â˜…
          </span>
        </div>
      </span>
    </div>
    <div
      className="user_rating"
    >
      Your Price: 
      <span
        className="price"
      >
        <div>
          <span>
            $
          </span>
          <span>
            $
          </span>
          <span>
            $
          </span>
        </div>
      </span>
    </div>
    <div
      className="user_rating"
    >
      Your Volume Level: 
      <span
        className="volume"
      >
        <span
          aria-label="volume level"
          role="img"
        >
          ðŸ“£
        </span>
      </span>
    </div>
  </div>
  <div
    className="user_content"
  >
    <p>
      Here's what you had to say
    </p>
    <p>
      Fusce posuere felis sed lacus. Morbi sem mauris, laoreet ut, rhoncus aliquet, pulvinar sed, nisl. Nunc rhoncus dui vel sem. Sed sagittis.
    </p>
  </div>
  <div
    className="userReview_controls"
  >
    <a
      href="/reviews/27"
      onClick={[Function]}
    >
      <button
        className="edit"
      >
        Edit Review
      </button>
    </a>
    <button
      className="delete"
      onClick={[Function]}
    >
      Delete Review
    </button>
  </div>
</li>
`;
